src/weapons.cpp

```diff
@@ -772,7 +783,7 @@ bool WeaponDistance::useWeapon(Player* player, Item* item, Creature* target) con
 	}
 
 	if (item->getWeaponType() == WEAPON_AMMO) {
-		Item* bow = player->getWeapon(true);
+		const auto& bow = player->getWeapon(true);
 		if (bow && bow->getHitChance() != 0) {
 			chance += bow->getHitChance();
 		}
```

Comment:
We can use `if` with `const auto&`
```suggestion
		if (const auto& bow = player->getWeapon(true)) {
			if (bow->getHitChance() != 0) {
				chance += bow->getHitChance();
			}
		}
```